### ISSUE TITLE
Review using Marshal.AllocHGlobal()

### ISSUE THREAD

**Issue by iSazonov** [PowerShell/PowerShell#15602]
.Net 6.0 Preview6 or Preview7 gets new API https://github.com/dotnet/runtime/issues/33244#issuecomment-862443475 (https://github.com/dotnet/runtime/pull/54006 and https://github.com/dotnet/runtime/issues/54297)

Perhaps we could replace Marshal.AllocHGlobal() (~62 usings in our code) with new API System.Runtime.InteropServices.NativeMemory.Alloc() to be faster ~50x with p/invokes.

**Comment by xtqqczze** [comment]
I don't think there are any differences in the memory alignment between `Marshal.AllocHGlobal` and `NativeMemory.Alloc`, see https://github.com/dotnet/runtime/issues/33228

So resolving this issue should be straightforward once next .Net preview is released.

**Comment by xtqqczze** [comment]
We can start work on this once #15896 is merged.

**Comment by xtqqczze** [comment]
@iSazonov I don't see where this is completed?

**Comment by xtqqczze** [comment]
If we split each change into a separate PR, we can likely get changes merged, with time. I think we could keep this issue open.
